js/viz/series/points/bar_point.js

```diff
@@ -45,6 +45,19 @@ export default _extend({}, symbolPoint, {
         }
     },
 
+    _calculateVisibility: function(x, y, width, height) {
+        const that = this;
+        const visibleArea = that._getVisibleArea();
+
+        if((visibleArea.minX >= (x + width) || (visibleArea.maxX <= x) ||
```

Comment:
We can just set `this.inVisibleArea` to this calculation. I think it will be more readable than this condition